catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/EmailSenderImpl.java

```diff
@@ -0,0 +1,194 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.alliance.nsili.endpoint;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EmailSenderImpl implements EmailSender {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSenderImpl.class);
+
+    private String fromEmail;
+
+    private String mailHost;
+
+    private String subject;
+
+    private String emailBody;
+
+    private String filename;
+
+    private int maxAttachmentSize;
+
+    /**
+     * EmailSenderImpl formats emails and sends them to the desired address using the fromEmail
+     * and through the desired mailHost.
+     *
+     * @param mailHost: non-null origin SMTP mailHost to send through
+     */
+    public EmailSenderImpl(String mailHost) {
+        notNull(mailHost, "mailHost must be non-null");
+
+        this.mailHost = mailHost;
+    }
+
+    /**
+     * sendEmail method sends email after receiving input parameters
+     *
+     * @param fromEmail: non-null origin address
+     * @param subject:   non-null subject line of email to be sent
+     * @param emailBody: non-null body of email to be sent
+     */
+    @Override
+    public void sendEmail(String fromEmail, String emailDest, String subject, String emailBody,
+            String filename, InputStream attachment) {
+        notNull(fromEmail, "fromEmail must be non-null");
+        notNull(subject, "subject must be non-null");
+        notNull(emailBody, "emailBody must be non-null");
+        notNull(filename, "filename must be non-null");
+
+        this.fromEmail = fromEmail;
+        this.subject = subject;
+        this.emailBody = emailBody;
+        this.filename = filename;
+
+        try {
+            InternetAddress emailAddr = new InternetAddress(emailDest);
+            emailAddr.validate();
+
+            Properties properties = createSessionProperies();
+            properties.put("mail.smtp.auth", "true");
+            properties.put("mail.smtp.starttls.enable", "true");
+            properties.put("mail.smtp.host", mailHost);
+            properties.put("mail.smtp.port", "25");
```

Comment:
This should be a parameter as well, as the port may be different depending on how the server is configured.